plugins/ksp/src/org/jetbrains/kotlin/ksp/processing/impl/ResolverImpl.kt

```diff
@@ -201,6 +199,15 @@ class ResolverImpl(
         } as FunctionDescriptor?
     }
 
+    fun resolvePropertyDeclaration(property: KSPropertyDeclaration): PropertyDescriptor? {
+        return when (property) {
+            is KSPropertyDeclarationImpl -> resolveDeclaration(property.ktProperty)
+            is KSPropertyDeclarationDescriptorImpl -> property.descriptor
+            is KSPropertyDeclarationJavaImpl -> null
```

Comment:
Sorry if there was any confusion in the previous comment, what I meant is that we should check for Java properties before trying to resolve a property, i.e. add checks before calling resolve for Java properties when doing override checking for properties.

Since this function`resolvePropertyDeclaration`'s name suggests it is for resolving a property to descriptor I would prefer to keep it as a general purpose function resolving properties. `resolveJavaDeclaration(property.psi)` actually won't work for properties in current implementation, but this is out of the scope of this PR, you can just leave a `TODO()` here other than a `null`.